### PR TITLE
Fix Exception on Nested TabBarView disposal

### LINKED ISSUE #18756: Exception while scrolling through TabBar navigation
```
Performing hot reload...
Reloaded 0 of 415 libraries in 866ms.
flutter: ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
flutter: The following assertion was thrown while finalizing the widget tree:
flutter: 'package:flutter/src/widgets/scroll_position.dart': Failed assertion: line 657 pos 12: 'pixels !=
flutter: null': is not true.
flutter:
flutter: Either the assertion indicates an error in the framework itself, or we should provide substantially
flutter: more information in this error message to help you determine and fix the underlying cause.
flutter: In either case, please report this assertion by filing a bug on GitHub:
flutter:   https://github.com/flutter/flutter/issues/new
flutter:
flutter: When the exception was thrown, this was the stack:
flutter: #2      ScrollPosition.dispose (package:flutter/src/widgets/scroll_position.dart)
flutter: #3      ScrollPositionWithSingleContext.dispose (package:flutter/src/widgets/scroll_position_with_single_context.dart:255:11)
flutter: #4      ScrollableState.dispose (package:flutter/src/widgets/scrollable.dart:324:14)
flutter: #5      StatefulElement.unmount (package:flutter/src/widgets/framework.dart:3821:12)
flutter: #6      _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1697:13)
flutter: #7      _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #8      ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #9      _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #10     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #11     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #12     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #13     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #14     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #15     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #16     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #17     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #18     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #19     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #20     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #21     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #22     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #23     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #24     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #25     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #26     MultiChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4742:16)
flutter: #27     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #28     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #29     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #30     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #31     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #32     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #33     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #34     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #35     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #36     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #37     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #38     SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4642:14)
flutter: #39     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #40     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #41     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #42     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #43     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #44     SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4642:14)
flutter: #45     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #46     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #47     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #48     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #49     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #50     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #51     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #52     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #53     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #54     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #55     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #56     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #57     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #58     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #59     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #60     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #61     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #62     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #63     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #64     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #65     SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4642:14)
flutter: #66     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #67     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #68     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #69     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #70     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #71     SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4642:14)
flutter: #72     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #73     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #74     SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4642:14)
flutter: #75     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #76     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #77     MultiChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4742:16)
flutter: #78     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #79     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #80     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #81     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #82     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #83     SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4642:14)
flutter: #84     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #85     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #86     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #87     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #88     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #89     SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4642:14)
flutter: #90     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #91     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #92     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #93     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #94     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #95     SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4642:14)
flutter: #96     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #97     _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #98     ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #99     _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #100    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #101    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #102    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #103    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #104    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #105    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #106    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #107    SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4642:14)
flutter: #108    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #109    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #110    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #111    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #112    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #113    SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4642:14)
flutter: #114    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #115    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #116    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #117    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #118    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #119    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #120    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #121    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #122    SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4642:14)
flutter: #123    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #124    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #125    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #126    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #127    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #128    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #129    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #130    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #131    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #132    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #133    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #134    MultiChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4742:16)
flutter: #135    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #136    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #137    _TheatreElement.visitChildren (package:flutter/src/widgets/overlay.dart:465:14)
flutter: #138    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #139    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #140    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #141    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #142    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #143    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #144    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #145    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #146    SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4642:14)
flutter: #147    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #148    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #149    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #150    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #151    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #152    SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4642:14)
flutter: #153    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #154    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #155    SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4642:14)
flutter: #156    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #157    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #158    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #159    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #160    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #161    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #162    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #163    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #164    SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4642:14)
flutter: #165    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #166    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #167    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #168    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #169    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #170    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #171    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #172    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #173    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #174    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #175    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #176    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #177    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #178    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #179    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #180    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #181    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #182    SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4642:14)
flutter: #183    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #184    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #185    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #186    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #187    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #188    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #189    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #190    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #191    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #192    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #193    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #194    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #195    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #196    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #197    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #198    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #199    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #200    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #201    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #202    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #203    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #204    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #205    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #206    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #207    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #208    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #209    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #210    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #211    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #212    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #213    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #214    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #215    SingleChildRenderObjectElement.visitChildren (package:flutter/src/widgets/framework.dart:4642:14)
flutter: #216    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #217    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #218    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #219    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #220    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #221    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #222    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #223    _InactiveElements._unmount.<anonymous closure> (package:flutter/src/widgets/framework.dart:1695:7)
flutter: #224    ComponentElement.visitChildren (package:flutter/src/widgets/framework.dart:3676:14)
flutter: #225    _InactiveElements._unmount (package:flutter/src/widgets/framework.dart:1693:13)
flutter: #226    ListIterable.forEach (dart:_internal/iterable.dart:39:13)
flutter: #227    _InactiveElements._unmountAll (package:flutter/src/widgets/framework.dart:1706:25)
flutter: #228    BuildOwner.finalizeTree.<anonymous closure> (package:flutter/src/widgets/framework.dart:2328:27)
flutter: #229    BuildOwner.lockState (package:flutter/src/widgets/framework.dart:2160:15)
flutter: #230    BuildOwner.finalizeTree (package:flutter/src/widgets/framework.dart:2327:7)
flutter: #231    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&RendererBinding&WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:628:18)
flutter: #232    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding&PaintingBinding&RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:208:5)
flutter: #233    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:990:15)
flutter: #234    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:930:9)
flutter: #235    _WidgetsFlutterBinding&BindingBase&GestureBinding&ServicesBinding&SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:842:5)
flutter: #236    _invoke (dart:ui/hooks.dart:120:13)
flutter: #237    _drawFrame (dart:ui/hooks.dart:109:3)
flutter: (elided 2 frames from class _AssertionError)
flutter: ════════════════════════════════════════════════════════════════════════════════════════════════════
```

### PR DESCRIPTION
**Pull request by shihaohong**
## Description

When a TabBarView is nested in a page of another TabBarView, there will be particular scenarios that will throw an exception. 

To reproduce this issue, two conditions must be true:
1) Tap on a tab that is adjacent to a tab containing a nested TabBarView
2) The nested tab has to be in between the initial tab and the tab that was tapped on 
(ie. current index = 0, tap on tab at index 3 when tab at index 2 has nested TabBarView)

This happens when the framework tries to build `_PagePosition`, then dispose of it before `applyViewportDimension` has the chance to be called to set `pixels` to a non-null value based on the size of the viewport. This fix adds a flag to determine if `applyViewportDimension` is invoked before `dispose`. If it hasn't been, then `dispose` will set `pixels` to an arbitrary value before attempting to dispose of it. 

First exception: This is due to the introduction of https://github.com/flutter/flutter/pull/29188/, which happens because the initial swap that is necessary for warping causes the nested page that is swapped but not shown to be disposed of before `applyViewportDimension` to be called. 

Second exception: This one is caused by the animation of the page during warp. The last frame does not build before the second `setState` is called to swap the children back to their original positions, causing the nested page to be built and disposed of before `applyViewportDimension` can be called.

TODO:
- [x] Add an issue to track adding a test to ensure that the warp animation for `TabBarView` builds its last frame before `setState` is called to re-swap the children back to their original locations. https://github.com/flutter/flutter/issues/32054
- [x] Track the potential need for a more robust swapping mechanism that doesn't cause pages that are not shown to be built and immediately disposed of. https://github.com/flutter/flutter/issues/32056

## Related Issues

Fixes https://github.com/flutter/flutter/issues/18756

## Tests

I added the following tests:

- A regression test to ensure that the exception does throw when a tab adjacent to nested tab is selected. It tests for the first exception case mentioned above. A separate issue will be created to track writing a test for the second exception case.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by chunhtai** [comment]
https://github.com/flutter/flutter/pull/31978 merged

**Review by HansMuller — APPROVED** [review]
This is definitely a workaround. Hopefully we'll be able to replace it with fixes for #32054, #32056.

File: packages/flutter/lib/src/widgets/page_view.dart [hunk 1]
```diff
@@ -251,6 +251,26 @@ class _PagePosition extends ScrollPositionWithSingleContext implements PageMetri
...
+  /// With [Tab]s, this happens when there are nested [TabBarView]s and there
+  /// is an attempt to warp over the nested tab to a tab adjacent to it.
+  ///
+  /// This flag will be set to true once the dimensions have been established
+  /// and [pixels] is set.
+  bool isInitialPixelsValueSet = false;
+
+  @override
+  void dispose() {
+    // Sets `pixels` to a non-null value before `ScrollPosition.dispose` is
```
Comment (chunhtai) [inline packages/flutter/lib/src/widgets/page_view.dart]: can you add a TODO and link to the issue and note once the issue is fix we can remove this workaround

Comment (shihaohong) [reply packages/flutter/lib/src/widgets/page_view.dart]: Done!

**Pull request merged**